packages/tpetra/core/test/inout/MatrixMarket_Tpetra_CrsMatrix_Dist_Binary.cpp

```diff
@@ -0,0 +1,822 @@
+    params.set("callFillComplete", true);
+    params.set("useTimers", true);
+
+    if(perProcess)
+      runTestPerProcess(testname, params);
+    else
+      runTestGlobal(testname, params);
+  }
+
+  //////////////////////////////
+  // Each test reads the binary file, applies, and compares
+  int runTestGlobal(
+    const std::string &testname,
+    Teuchos::ParameterList &params
+  )
+  {
+    if (comm->getRank() == 0) 
+      std::cout << "\n\nBEGIN GLOBAL TEST " << testname << "\n" << std::endl;
+
+    // Read the binary file
+    params.set("binary", true);  // This invokes the binary reader
+    Teuchos::RCP<Tpetra::Distribution<gno_t, scalar_t> > distTest;  // Not used
+    Teuchos::RCP<matrix_t> AmatTest = readFile(binfilename, testname, params, distTest);
+
+    // Apply and compare
+    Teuchos::RCP<vector_t> yvec = applyMatrix(testname, *AmatTest);
+    return compareToBaseline(testname, yvec);
+  }
+
+  //////////////////////////////
+  // Each test reads the mtx file, creates a distributed matrix AmatWriter,
+  //  writes the per-process binary files using AmatWriter, reads the per-process
+  //  files, applies, and compares
+  int runTestPerProcess(
+    const std::string &testname,
+    Teuchos::ParameterList &params
+  )
+  {
+    if (comm->getRank() == 0) 
+      std::cout << "\n\nBEGIN PER-PROCESS TEST " << testname << "\n" << std::endl;
+
+    // Create AmatWriter by reading the input mtx file with the required distribution
+    Teuchos::RCP<Tpetra::Distribution<gno_t, scalar_t> > distWriter;  // Not used
+    Teuchos::RCP<matrix_t> AmatWriter = readFile(filename, testname, params, distWriter);
+
+    // Write the binary files using AmatWriter
+    writeBinaryPerProcess(testname, AmatWriter);
+
+    // Read the binary files
+    params.set("binary", true);  // This invokes binary readers.
+    params.set("readPerProcess", true);
+    Teuchos::RCP<Tpetra::Distribution<gno_t, scalar_t> > distTest;  // Not used
+    Teuchos::RCP<matrix_t> AmatTest = readFile(binfilename, testname, params, distTest);
+
+    // Clean-up the per-process binary files
+    // cleanBinaryPerProcess();
+
+    // Apply and compare
+    Teuchos::RCP<vector_t> yvec = applyMatrix(testname, *AmatTest);
+    return compareToBaseline(testname, yvec);
+  }
+
+  //////////////////////////////
+  // Each test reads, writes binary file(s), reads the binary file(s), applies, and compares
+  // This test should not compare the current results with baseline's results,
+  // because the LTB distribution only keeps the lower triangular entries.
+  // Therefore, this test compares the current results with AmatWriter's results.
+  int runTestLTB(
+    const std::string &testname,
+    Teuchos::ParameterList &params
+  )
+  {
+    params.set("verbose", true);
+    params.set("callFillComplete", true);
+    params.set("useTimers", true);
+
+    if(perProcess)
+      runTestLTBPerProcess(testname, params);
+    else
+      runTestLTBGlobal(testname, params);
+  }
```

Comment:
Function runTestLTB needs to return the error code from runTestPerProcess or runTestGlobal.

Reply:
Added the missing returns and enabled the tests back.